src/utils.jl

```diff
@@ -66,3 +66,88 @@ function check_pkg_version(pkg::String, ver::String;
            pkg_info[pkg].version >= VersionNumber(ver) :
            pkg_info[pkg].version > VersionNumber(ver)
 end
+
+
+# RetCode handling for BBO and others.
+using Logging
+ # Define the ReturnCode type using @enum
+@enum ReturnCode begin
+    Default
+    Success
+    Terminated
+    MaxIters
+    MaxTime
+    DtLessThanMin
+    Unstable
+    InitialFailure
+    ConvergenceFailure
+    Failure
+    Infeasible
+end
+ 
+# Define a dictionary to map regular expressions to ReturnCode values
+const STOP_REASON_MAP = Dict(
```

Comment:
Since it's only BBO specific move it to there, create a utils.jl file in it

Reply:
Even though currently it is only BBO specific we can update the Dictionary for other solvers as we go and whenever needed so easier maintenance that is why I wrote it here.

Reply:
I see, makes sense 👍